book/example_8_chat_server.md

```diff
@@ -215,15 +225,16 @@ flowchart TB
 ```
 
 ## client.rs
-The chat client is not very different from the echo server client, with two small additions and two 
-new design patterns. The additions are 
-(1) it has a Rust helper routine `pretty_print_msg` for formatting output, and (2) it sends a
-`ConnectRequest` message to the server upon invocation. The new design patterns are (a) a degenerate `source_iter` pipeline that runs once 
-as a "bootstrap" in the first epoch, and (b) the use of
-`cross_join` as a "gated buffer" to postpone sending messages.
+The chat client is not very different from the echo server client, with two new design patterns:
+ 1. a degenerate `source_iter` pipeline that runs once 
+as a "bootstrap" in the first epoch
```

Comment:
I guess this should eventually be its own `init` operator

Maybe once we have some sort of pseudo-ops